tests/integration/test_network.py

```diff
@@ -1,12 +1,24 @@
 from pytest_docker_tools import network
 
 test_network_1 = network()
+test_network_2 = network()
 
 
-def test_network_created(docker_client, test_network_1):
+def test_network_1_created(docker_client, test_network_1):
     for n in docker_client.networks.list():
         if n.id == test_network_1.id:
             # Looks like we managed to start one!
             break
     else:
         assert False, "Looks like we failed to create a network"
+
+
+def test_reusable_network_2_created(
+    enable_container_reuse, docker_client, test_network_2
+):
+    for n in docker_client.networks.list():
+        if n.id == test_network_2.id:
+            # Looks like we managed to start one!
+            break
+    else:
+        assert False, "Looks like we failed to create a network"
```

Comment:
I think we should merge this and do a release, but then if you have time can we iterate on some more tests?

Looking at this test as an example, it doesn't verify the `--reuse-containers` works, just that a network does exist. Especially as I probably won't use this feature myself I want to make sure I don't introduce regressions for it.

I think we should be using either https://docs.pytest.org/en/6.2.x/reference.html#testdir or https://docs.pytest.org/en/6.2.x/reference.html#std-fixture-pytester  (see example of use here https://github.com/aio-libs/aiohttp/blob/master/tests/test_pytest_plugin.py) to do multiple pytest runs and verifying the state before, during and after.

E.g. for the reusable container tests the test should run a nested pytest with reusable-containers on and verify that the container is still running (using the raw docker api (`docker_client` fixture)). Then it should run pytest again and verify that there is still only a single instance of the container. And we should verify the id hasn't changed either. The test should assert about the outcomes too (`testdir.runpytest().assert_outcomes(passed=1)`). And we should clean up the leftovers after the 2nd test result.

I'd like a test like that for containers, volumes and networks.

And then maybe a similar test but that asserts what happens if you run once with `--reusable-containers` and then straight away again without the flag. I'm not too worried about what it does there (as long as it doesn't blow up), but I would like to lock in the behaviour to avoid accidentally changing it in future.

Does that make sense? Is it something you can help with? If so i'll get a release out this morning.

Reply:
I agree, I also was wondering on how a test for conditions and exceptions inside the fixtures should look like and I also came across pytester. Also I agree with doing a release now, as the feature is currently broken if the default bridge isn't used.

I'm happy to help with adding nested tests. However this might take a bit of time since there are a few other things I need to deal with. I hope this is okay.